core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleWriter.java

```diff
@@ -145,6 +159,12 @@ public void startRDF() throws RDFHandlerException {
 			xsdStringToPlainLiteral = getWriterConfig().get(BasicWriterSettings.XSD_STRING_TO_PLAIN_LITERAL);
 			prettyPrint = getWriterConfig().get(BasicWriterSettings.PRETTY_PRINT);
 			inlineBNodes = getWriterConfig().get(BasicWriterSettings.INLINE_BLANK_NODES);
+
+			if (isBuffering()) {
+				this.bufferedStatements = getModelFactory().createEmptyModel();
+				this.bufferSize = inlineBNodes ? -1 : DEFAULT_BUFFER_SIZE;
```

Comment:
Btw. If you use a `long` for bufferSize you can just set it to be `Long.MAX_VALUE` and you don't have to deal with -1 as a special case. Dunno if it would help, but it's a decent approach since you would run out of memory way before reaching that limit (minimum 4 611 686  terabytes).

Reply:
Good idea.